helix-vcs/src/git.rs

```diff
@@ -61,6 +74,28 @@ impl Git {
 
         Ok(res)
     }
+
+    /// Emulates the result of running `git status` from the command line.
+    fn status(repo: &Repository) -> Result<Box<dyn Iterator<Item = Result<FileChange>>>> {
+        let work_dir = repo
+            .work_dir()
+            .ok_or_else(|| anyhow::anyhow!("working tree not found"))?
+            .to_path_buf();
+
+        // Here we discard the `status.showUntrackedFiles` config, as it makes little sense in our
+        // case to not list new (untracked) files. We could have respected this config if the
+        // default value weren't `Collapsed` though, as this default value would render the feature
+        // unusable to many.
+        let status_platform = repo
+            .status(gix::progress::Discard)?
+            .untracked_files(UntrackedFiles::Files)
+            .index_worktree_rewrites(Some(Default::default()));
+
+        Ok(Box::new(StatusIter {
+            work_dir,
+            inner: status_platform.into_index_worktree_iter(Default::default())?,
```

Comment:
I don't like the `Default` there as it's not clear what it stands for, but also think that `vec![]` or similar would be much more descriptive.
That's probably an area where the API can improve, even though I don't know how without making it more complicated.

With that said, maybe a variable like `let empty_pathspec = vec![]` would do a slightly better job here (even though one would still have to know that this means there is no restriction.

Reply:
Not 100% relevant, but I remember checking this part of the source and wondering why a generic `IntoIterator` param isn't used. I mean this method immediately calls `index_worktree_status`, which indeed takes a generic item impl. Here we're forced to do an allocation even though we're just passing in an empty vec. Negligible though.

Reply:
Now explicitly naming the argument with comment.

Reply:
Thanks for the hint! I think I did it to allow passing empty items easily, but now it's `IntoIterator<Item = BString>` which at least allows to pass `None` as well, a little easier.